src/typechecker.erl

```diff
@@ -3933,6 +3983,45 @@ is_power_of_two(N) when N rem 2 == 0 ->
     is_power_of_two(N div 2);
 is_power_of_two(_) -> false.
 
+-spec union_var_binds_symmetrical(list(), #tenv{}) -> map().
+union_var_binds_symmetrical([], TEnv) ->
+    #{};
+union_var_binds_symmetrical(VarBindsList, TEnv) ->
+    Lub = fun(_K, Ty1, Ty2) ->
+        UTy = type(union, [Ty1, Ty2]),
+        NTy = normalize(UTy, TEnv),
+        NTy
+    end,
+    union_var_binds_symmetrical_help(VarBindsList, Lub).
+
+-spec union_var_binds_symmetrical_help(list(), function()) -> map().
```

Comment:
Please use `union_var_binds_help` here instead of defining a new function. You can just pass the `LUB` function to `union_var_binds_help`.

Reply:
No. This was done initially, but it doesn't work. See my comment on previous patch.

Reply:
I can't find the comment you're referring to. How is this function different from `union_var_binds_help`?

Reply:
https://github.com/josefs/Gradualizer/pull/241#discussion_r363033009

Reply:
Basically, instead of merging all var binds, we merge them when they are present in every var binds.

```
union_var_binds : [{ x: integer }, {x: integer, y: integer}] = {x: integer, y: integer}
union_var_binds_symmetrical : [{ x: integer }, {x: integer, y: integer}] = {x: integer}
```

Reply:
I see. That makes sense.